diff --git a/integration/scripts/files/elements/ubuntu-mysql/pre-install.d/10-percona-apt-key b/integration/scripts/files/elements/ubuntu-mysql/pre-install.d/10-percona-apt-key
index d62c605..77304f2 100755
--- a/integration/scripts/files/elements/ubuntu-mysql/pre-install.d/10-percona-apt-key
+++ b/integration/scripts/files/elements/ubuntu-mysql/pre-install.d/10-percona-apt-key
@@ -9,6 +9,9 @@ set -o xtrace
 [ -n "${GUEST_USERNAME}" ] || die "GUEST_USERNAME needs to be set to the user for the guest image"
 [ -n "${RELEASE}" ] || die "RELEASE must be set to a valid Ubuntu release (e.g. trusty)"
 
+# This website does not provide ppc64le binaries. The bug below is fixed in Xenial
+exit 0
+
 # Add Percona GPG key
 mkdir -p /home/${GUEST_USERNAME}/.gnupg
 
diff --git a/integration/scripts/files/elements/ubuntu-trusty-mysql/install.d/30-mysql b/integration/scripts/files/elements/ubuntu-trusty-mysql/install.d/30-mysql
index 887bda2..da2a9d4 100755
--- a/integration/scripts/files/elements/ubuntu-trusty-mysql/install.d/30-mysql
+++ b/integration/scripts/files/elements/ubuntu-trusty-mysql/install.d/30-mysql
@@ -8,14 +8,7 @@ set -o xtrace
 
 export DEBIAN_FRONTEND=noninteractive
 
-# The fix to make versions of percona-xtrabackup > v2.2 work with Trove
-# was put into the mysql guestagent code for Mitaka. There are no current
-# plans to backport so we need to make sure the guest generated when the
-# tests are run for Kilo or Liberty get the 2.2 verson of PXB
-if [[ $BRANCH_OVERRIDE == "stable/kilo" || $BRANCH_OVERRIDE == "stable/liberty" ]]; then
-    PXB_VERSION_OVERRIDE="-22"
-fi
-apt-get --allow-unauthenticated -y install libmysqlclient18 mysql-server-5.6 percona-xtrabackup${PXB_VERSION_OVERRIDE}
+apt-get --allow-unauthenticated -y install libmysqlclient18 mysql-server percona-xtrabackup
 
 cat >/etc/mysql/conf.d/no_perf_schema.cnf <<_EOF_
 [mysqld]
diff --git a/integration/scripts/files/elements/ubuntu-xenial-mysql/install.d/30-mysql b/integration/scripts/files/elements/ubuntu-xenial-mysql/install.d/30-mysql
index 75ccdc6..a18206d 100755
--- a/integration/scripts/files/elements/ubuntu-xenial-mysql/install.d/30-mysql
+++ b/integration/scripts/files/elements/ubuntu-xenial-mysql/install.d/30-mysql
@@ -8,10 +8,31 @@ set -o xtrace
 
 export DEBIAN_FRONTEND=noninteractive
 
-add-apt-repository 'deb http://archive.ubuntu.com/ubuntu trusty universe'
 apt-get -y  update
 
-apt-get --allow-unauthenticated -y install mysql-client-5.6 mysql-server-5.6 percona-xtrabackup${PXB_VERSION_OVERRIDE}
+# Xenial provides mysql 5.7 which requires percona-xtrabackup-24 which needs
+# to be compiled from source code as the percona website does not support power
+
+apt-get --allow-unauthenticated -y install build-essential flex bison automake autoconf \
+   libtool cmake libaio-dev mysql-client libncurses-dev zlib1g-dev \
+   libgcrypt11-dev libev-dev libcurl4-gnutls-dev vim-common
+
+pushd /tmp
+
+git clone https://github.com/percona/percona-xtrabackup.git
+cd percona-xtrabackup
+git checkout 2.4
+
+mkdir /tmp/boost
+cmake -DDOWNLOAD_BOOST=1 -DWITH_BOOST=/tmp/boost -DBUILD_CONFIG=xtrabackup_release -DWITH_MAN_PAGES=OFF && make -j4
+make install
+ln -s /usr/local/xtrabackup/bin/* /usr/bin/
+
+popd
+
+rm -rf /tmp/boost /tmp/percona-xtrabackup
+
+apt-get --allow-unauthenticated -y install mysql-client mysql-server
 
 cat >/etc/mysql/conf.d/no_perf_schema.cnf <<_EOF_
 [mysqld]
@@ -26,7 +47,9 @@ cat >/etc/mysql/my.cnf <<_EOF_
 !includedir /etc/mysql/conf.d/
 _EOF_
 
-rm /etc/init/mysql.conf
-systemctl daemon-reload
+if [ -e /etc/init/mysql.conf ]; then
+    rm -f /etc/init/mysql.conf
+fi
+
 systemctl enable mysql
 
