diff --git a/integration/scripts/files/elements/ubuntu-mongodb/install.d/20-mongodb b/integration/scripts/files/elements/ubuntu-mongodb/install.d/20-mongodb
index 4ab5c34..bf4a5b0 100755
--- a/integration/scripts/files/elements/ubuntu-mongodb/install.d/20-mongodb
+++ b/integration/scripts/files/elements/ubuntu-mongodb/install.d/20-mongodb
@@ -1,8 +1,62 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 set -o xtrace
 
 export DEBIAN_FRONTEND=noninteractive
 
-apt-get --allow-unauthenticated -y install mongodb-org=3.2.6
+MONGODB_CONF_DIR=/etc
+MONGODB_CONF_NAME=mongod.conf
+MONGODB_CONF=$MONGODB_CONF_DIR/$MONGODB_CONF_NAME
+MONGOS_CONF_NAME=mongos.conf
+MONGOS_CONF=$MONGODB_CONF_DIR/$MONGOS_CONF_NAME
+
+ESC_MONGODB_CONF="${MONGODB_CONF//\//\\/}"
+ESC_MONGOS_CONF="${MONGOS_CONF//\//\\/}"
+
+if [ "$DIB_MYENTERPRISE_EDITION" == "true" ]; then
+    apt-get install -y $DIB_MYDBPKG
+    exit 0
+fi
+
+if [ "$DIB_MYCOMMUNITY_EDITION" == "false" ]; then
+    exit 0
+fi
+
+# Install build and runtime prerequisites
+apt-get --allow-unauthenticated install -y build-essential scons tar wget \
+    libcurl3 libgssapi-krb5-2 libldap-2.4-2 libpcap0.8 libpci3 libsasl2-2 \
+    libsensors4 libsnmp30 libssl1.0.0 libwrap0 libboost-filesystem-dev \
+    libboost-program-options-dev libboost-system-dev libboost-thread-dev
+
+# Build and install mongodb from source code from mongodb.org
+
+ARCH_SAV=$ARCH
+MONGODB_NAME=$DIB_MYDBSRCPKG
+
+cd /tmp
+wget https://fastdl.mongodb.org/src/$MONGODB_NAME.tar.gz
+tar xvzf $MONGODB_NAME.tar.gz
+cd $MONGODB_NAME
+
+unset ARCH
+scons --disable-warnings-as-errors core
+scons --prefix=/usr install
+export ARCH=$ARCH_SAV
+
+adduser --system --group --no-create-home mongodb
+
+# Create the data dirs
+
+MONGODB_DATA_DIR=/var/lib/mongodb
+MONGODB_LOG_DIR=/var/log/mongodb
+
+mkdir -p $MONGODB_DATA_DIR
+chown -R mongodb:mongodb $MONGODB_DATA_DIR
+chmod 750 $MONGODB_DATA_DIR
+mkdir -p $MONGODB_LOG_DIR
+chown -R mongodb:mongodb $MONGODB_LOG_DIR
+
+# Clean the build tree to minimize the size of the image and to avoid running out of space when creating the image
+rm /tmp/$MONGODB_NAME.tar.gz
+rm -rf /tmp/$MONGODB_NAME
diff --git a/integration/scripts/files/elements/ubuntu-mongodb/install.d/25-trove-mongo-dep b/integration/scripts/files/elements/ubuntu-mongodb/install.d/25-trove-mongo-dep
index a8b3ddc..6774019 100755
--- a/integration/scripts/files/elements/ubuntu-mongodb/install.d/25-trove-mongo-dep
+++ b/integration/scripts/files/elements/ubuntu-mongodb/install.d/25-trove-mongo-dep
@@ -6,4 +6,9 @@
 set -e
 set -o xtrace
 
-pip2 install pymongo>=3.0.2,!=3.1
+pip2 install 'pymongo>=3.0.2,!=3.1'
+
+if [ "$DIB_MYCOMMUNITY_EDITION" == "true" ]; then
+    # This is the registered package to prevent Trove from re-installing mongodb
+    apt-get --allow-unauthenticated install -y python-pymongo-doc
+fi
diff --git a/integration/scripts/files/elements/ubuntu-mongodb/install.d/30-mongodb-conf b/integration/scripts/files/elements/ubuntu-mongodb/install.d/30-mongodb-conf
index afb5828..42cd816 100755
--- a/integration/scripts/files/elements/ubuntu-mongodb/install.d/30-mongodb-conf
+++ b/integration/scripts/files/elements/ubuntu-mongodb/install.d/30-mongodb-conf
@@ -1,26 +1,45 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 set -o xtrace
 
-# Remove the default pid file
-rm -f /var/run/mongodb.pid
+if [ -e /etc/mongod.conf ]; then
+    cp /etc/mongod.conf /etc/mongod.conf.orig
+fi
 
+if [ "$DIB_MYDEBUG" == "true" ]; then
+    DEBUGLEVEL=3                  # Max is 5
+else
+    DEBUGLEVEL=0                  # Default informational
+fi
 
 cat > /etc/mongod.conf << '_EOF_'
+processManagement.pidFilePath: /var/run/mongodb/mongodb.pid
 storage.dbPath: /var/lib/mongodb
 security.authorization: enabled
 storage.engine: wiredTiger
 storage.journal.enabled: true
+systemLog.verbosity: DEBUGLEVEL
+systemLog.component.accessControl.verbosity: 1
 systemLog.destination: file
 systemLog.logAppend: true
 systemLog.path: /var/log/mongodb/mongod.log
 _EOF_
 
+sed -i "s/DEBUGLEVEL/$DEBUGLEVEL/" /etc/mongod.conf
 
+if [ -e /etc/mongos.conf ]; then
+    cp /etc/mongos.conf /etc/mongos.conf.orig
+fi
+
+# TBD.  Mongos is not started.  Will be required for sharded clusters
 cat > /etc/mongos.conf << '_EOF_'
-security.authorization: enabled
+sharding.configDB: clustering/cfg1.example.net:27017
+storage.engine: wiredTiger
 systemLog.destination: file
 systemLog.logAppend: true
+systemLog.verbosity: DEBUGLEVEL
 systemLog.path: /var/log/mongodb/mongos.log
 _EOF_
+
+sed -i "s/DEBUGLEVEL/$DEBUGLEVEL/" /etc/mongos.conf
diff --git a/integration/scripts/files/elements/ubuntu-mongodb/pre-install.d/10-mongodb-apt-key b/integration/scripts/files/elements/ubuntu-mongodb/pre-install.d/10-mongodb-apt-key
index 1345f50..59716fd 100755
--- a/integration/scripts/files/elements/ubuntu-mongodb/pre-install.d/10-mongodb-apt-key
+++ b/integration/scripts/files/elements/ubuntu-mongodb/pre-install.d/10-mongodb-apt-key
@@ -1,14 +1,28 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 set -o xtrace
 
-[ -n "${RELEASE}" ] || die "RELEASE must be set to either Precise or Quantal"
+[ -n "${RELEASE}" ] || die "RELEASE must be set to either trusty or xenial"
 
-apt-get --allow-unauthenticated -y install software-properties-common
+if [ "$DIB_MYCOMMUNITY_EDITION" == "true" ]; then
+    # For ppc64le, we build and install from source code as there are no ppc64le packages at the community website
+    exit 0
+fi
 
-apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv EA312927
+if [ "$DIB_MYENTERPRISE_EDITION" == "true" ]; then
+    apt-get --allow-unauthenticated -y install software-properties-common
 
-echo "deb http://repo.mongodb.org/apt/ubuntu "$(lsb_release -sc)"/mongodb-org/3.2 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.2.list
+    if [ "$DIB_MYDBVERSION" == "3.3" ]; then
+        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 0C49F3730359A14518585931BC711F9BA15703C6
+    elif [ "$DIB_MYDBVERSION" == "3.4" ]; then
+        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 0C49F3730359A14518585931BC711F9BA15703C6
+    else
+        echo "ERROR invalid Mongodb Enterprise version $DIB_MYDBVERSION"
+        exit 1
+    fi
 
-apt-get -y update
+    echo "deb http://repo.mongodb.com/apt/ubuntu "$(lsb_release -sc)"/mongodb-enterprise/$DIB_MYDBVERSION multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-enterprise.list
+
+    apt-get -y update
+fi
diff --git a/integration/scripts/files/elements/ubuntu-xenial-mongodb/element-deps b/integration/scripts/files/elements/ubuntu-xenial-mongodb/element-deps
new file mode 100755
index 0000000..6a5964e
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-xenial-mongodb/element-deps
@@ -0,0 +1 @@
+ubuntu-mongodb
diff --git a/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/10-mongodb-thp b/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/10-mongodb-thp
new file mode 100755
index 0000000..3229917
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/10-mongodb-thp
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -e
+set -o xtrace
+
+cat > "/etc/rc.local" << _EOF_
+# Make sure to disable Linux kernel feature transparent huge pages,
+# it will affect greatly both memory usage and latency in a negative way.
+if test -f /sys/kernel/mm/transparent_hugepage/defrag; then
+  echo never > /sys/kernel/mm/transparent_hugepage/defrag
+fi
+if test -f /sys/kernel/mm/transparent_hugepage/enabled; then
+  echo never > /sys/kernel/mm/transparent_hugepage/enabled
+fi
+
+_EOF_
+
diff --git a/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/41-mongod-systemd b/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/41-mongod-systemd
new file mode 100755
index 0000000..db3a6b8
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/41-mongod-systemd
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -e
+set -o xtrace
+
+cat > /lib/systemd/system/mongod.service << '_EOF_'
+[Unit]
+Description=High-performance, schema-free document-oriented database
+After=network.target
+Documentation=https://docs.mongodb.org/manual
+
+[Service]
+User=mongodb
+Group=mongodb
+RuntimeDirectory=mongodb
+ExecStart=/usr/bin/mongod --config /etc/mongod.conf
+
+# file size
+LimitFSIZE=infinity
+# cpu time
+LimitCPU=infinity
+# virtual memory size
+LimitAS=infinity
+# open files
+LimitNOFILE=64000
+# processes/threads
+LimitNPROC=64000
+# memlock
+LimitMEMLOCK=2000000
+# total threads (user+kernel)
+TasksMax=infinity
+TasksAccounting=false
+
+# Recommended limits for for mongod as specified in
+# http://docs.mongodb.org/manual/reference/ulimit/#recommended-settings
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+
+systemctl enable mongod.service
diff --git a/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/42-mongos-systemd b/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/42-mongos-systemd
new file mode 100755
index 0000000..a8d2761
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-xenial-mongodb/install.d/42-mongos-systemd
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+set -e
+set -o xtrace
+
+cat > /lib/systemd/system/mongos.service << '_EOF_'
+[Unit]
+Description=High-performance, schema-free document-oriented database
+After=network.target
+Documentation=https://docs.mongodb.org/manual
+
+[Service]
+User=mongodb
+Group=mongodb
+RuntimeDirectory=mongodb
+ExecStart=/usr/bin/mongos --config /etc/mongos.conf
+
+# file size
+LimitFSIZE=infinity
+# cpu time
+LimitCPU=infinity
+# virtual memory size
+LimitAS=infinity
+# open files
+LimitNOFILE=64000
+# processes/threads
+LimitNPROC=64000
+# memlock
+LimitMEMLOCK=2000000
+# total threads (user+kernel)
+TasksMax=infinity
+TasksAccounting=false
+
+# Recommended limits for for mongod as specified in
+# http://docs.mongodb.org/manual/reference/ulimit/#recommended-settings
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+
+# TBD.  Started for sharded clusters
+# systemctl enable mongos.service
diff --git a/integration/scripts/files/elements/ubuntu-mongodb/install.d/10-mongodb-thp b/integration/scripts/files/elements/ubuntu-mongodb/install.d/10-mongodb-thp
deleted file mode 100755
index 48d6c8b..0000000
--- a/integration/scripts/files/elements/ubuntu-mongodb/install.d/10-mongodb-thp
+++ /dev/null
@@ -1,42 +0,0 @@
-#!/bin/sh
-
-set -e
-set -o xtrace
-
-export DEBIAN_FRONTEND=noninteractive
-
-cat > /etc/init.d/disable-transparent-hugepages << '_EOF_'
-#!/bin/sh
-### BEGIN INIT INFO
-# Provides:          disable-transparent-hugepages
-# Required-Start:    $local_fs
-# Required-Stop:
-# X-Start-Before:    mongod mongodb-mms-automation-agent
-# Default-Start:     2 3 4 5
-# Default-Stop:      0 1 6
-# Short-Description: Disable Linux transparent huge pages
-# Description:       Disable Linux transparent huge pages, to improve
-#                    database performance.
-### END INIT INFO
-
-case $1 in
-  start)
-    if [ -d /sys/kernel/mm/transparent_hugepage ]; then
-      thp_path=/sys/kernel/mm/transparent_hugepage
-    elif [ -d /sys/kernel/mm/redhat_transparent_hugepage ]; then
-      thp_path=/sys/kernel/mm/redhat_transparent_hugepage
-    else
-      return 0
-    fi
-
-    echo 'never' > ${thp_path}/enabled
-    echo 'never' > ${thp_path}/defrag
-
-    unset thp_path
-    ;;
-esac
-_EOF_
-
-chmod 755 /etc/init.d/disable-transparent-hugepages
-
-update-rc.d disable-transparent-hugepages defaults
diff --git a/integration/scripts/files/elements/ubuntu-mongodb/install.d/41-mongod-init b/integration/scripts/files/elements/ubuntu-mongodb/install.d/41-mongod-init
deleted file mode 100755
index 77a1292..0000000
--- a/integration/scripts/files/elements/ubuntu-mongodb/install.d/41-mongod-init
+++ /dev/null
@@ -1,46 +0,0 @@
-#!/bin/sh
-
-set -e
-set -o xtrace
-
-cat > /etc/init/mongod.conf << '_EOF_'
-limit fsize unlimited unlimited
-limit cpu unlimited unlimited
-limit as unlimited unlimited
-limit nofile 64000 64000
-limit rss unlimited unlimited
-limit nproc 64000 64000
-
-kill timeout 300 # wait 300s between SIGTERM and SIGKILL.
-
-pre-start script
-  mkdir -p /var/run/mongodb/
-  touch /var/run/mongodb/mongod.pid
-  chown mongodb -R /var/run/mongodb/
-end script
-
-start on runlevel [2345]
-stop on runlevel [06]
-
-script
-  CONF=/etc/mongod.conf
-
-  # Handle NUMA access to CPUs (SERVER-3574)
-  # This verifies the existence of numactl as well as testing that the command works
-  NUMACTL_ARGS="--interleave=all"
-  if which numactl >/dev/null 2>/dev/null && numactl $NUMACTL_ARGS ls / >/dev/null 2>/dev/null
-  then
-    NUMACTL="$(which numactl) -- $NUMACTL_ARGS"
-    DAEMON_OPTS=${DAEMON_OPTS:-"--config $CONF"}
-  else
-    NUMACTL=""
-    DAEMON_OPTS="-- "${DAEMON_OPTS:-"--config $CONF"}
-  fi
-
-  exec start-stop-daemon --start \
-        --chuid mongodb \
-        --pidfile /var/run/mongod.pid \
-        --make-pidfile \
-        --exec $NUMACTL /usr/bin/mongod $DAEMON_OPTS
-end script
-_EOF_
diff --git a/integration/scripts/files/elements/ubuntu-mongodb/install.d/42-mongos-init b/integration/scripts/files/elements/ubuntu-mongodb/install.d/42-mongos-init
deleted file mode 100755
index dc91fdb..0000000
--- a/integration/scripts/files/elements/ubuntu-mongodb/install.d/42-mongos-init
+++ /dev/null
@@ -1,30 +0,0 @@
-#!/bin/sh
-
-set -e
-set -o xtrace
-
-cat > /etc/init/mongos.conf << '_EOF_'
-limit fsize unlimited unlimited
-limit cpu unlimited unlimited
-limit as unlimited unlimited
-limit nofile 64000 64000
-limit rss unlimited unlimited
-limit nproc 64000 64000
-
-pre-start script
-  mkdir -p /var/run/mongodb/
-  touch /var/run/mongodb/mongos.pid
-  chown mongodb -R /var/run/mongodb/
-end script
-
-start on runlevel [2345]
-stop on runlevel [06]
-
-script
-    exec start-stop-daemon --start \
-        --chuid mongodb \
-        --pidfile /var/run/mongos.pid \
-        --make-pidfile \
-        --exec  /usr/bin/mongos -- --config /etc/mongos.conf
-end script
-_EOF_
