diff --git a/integration/scripts/files/elements/ubuntu-guest/install.d/29-user b/integration/scripts/files/elements/ubuntu-guest/install.d/29-user
new file mode 100755
index 0000000..99f6896
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-guest/install.d/29-user
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# CONTEXT: GUEST during CONSTRUCTION as ROOT
+# PURPOSE: Add the guest image user that will own the trove agent source...if the user does not already exist
+
+set -e
+set -o xtrace
+
+if ! id -u ${GUEST_USERNAME} >/dev/null 2>&1; then
+    echo "Adding ${GUEST_USERNAME} user"
+    useradd -G sudo -m ${GUEST_USERNAME} -s /bin/bash
+    chown ${GUEST_USERNAME}:${GUEST_USERNAME} /home/${GUEST_USERNAME}
+    passwd ${GUEST_USERNAME} <<_EOF_
+${GUEST_USERNAME}
+${GUEST_USERNAME}
+_EOF_
+fi
+
diff --git a/integration/scripts/files/elements/ubuntu-guest/install.d/70-trove-debug b/integration/scripts/files/elements/ubuntu-guest/install.d/70-trove-debug
new file mode 100755
index 0000000..128b7b8
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-guest/install.d/70-trove-debug
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -e
+set -o xtrace
+
+# CONTEXT: GUEST during CONSTRUCTION as ROOT
+# PURPOSE: Enables LOG.debug() in trove-guestagent
+
+if [ "$DIB_MYDEBUG" == "false" ]; then
+    exit 0
+fi
+
+# This command will be conditionally invoked during boot from systemd or upstart
+
+cat > /home/$GUEST_USERNAME/trove-guest-enable-debug.sh << _EOF_
+#!/bin/bash
+
+sed -i "s/^debug =.*/debug = True/" /etc/trove/conf.d/trove-guestagent.conf
+
+_EOF_
+
+chown $GUEST_USERNAME:root /home/$GUEST_USERNAME/trove-guest-enable-debug.sh
+chmod 750 /home/$GUEST_USERNAME/trove-guest-enable-debug.sh
diff --git a/integration/scripts/files/elements/ubuntu-trusty-guest/extra-data.d/70-trove-guestagent b/integration/scripts/files/elements/ubuntu-trusty-guest/extra-data.d/70-trove-guestagent
new file mode 100755
index 0000000..85233a9
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-trusty-guest/extra-data.d/70-trove-guestagent
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
+set -o xtrace
+
+# CONTEXT: HOST prior to IMAGE BUILD as SCRIPT USER
+# PURPOSE: stages the trove github repo and trove guestagent conf so that
+#          these items can be placed in the image so that systemd does not
+#          rsync them from a controller
+
+source $_LIB/die
+
+[ -n "$TMP_HOOKS_PATH" ] || die "Temp hook path not set"
+
+[ -n "${HOST_USERNAME}" ] || die "HOST_USERNAME needs to be set to the user for the current user on the host"
+[ -n "${PATH_TROVE}" ] || die "PATH_TROVE needs to be set to the path to the trove directory on the trovestack host"
+
+cp /home/${HOST_USERNAME}/trove-guestagent.conf ${TMP_HOOKS_PATH}
+cp -r ${PATH_TROVE} ${TMP_HOOKS_PATH}
+
diff --git a/integration/scripts/files/elements/ubuntu-trusty-guest/install.d/20-etc b/integration/scripts/files/elements/ubuntu-trusty-guest/install.d/20-etc
index 8ac19f7..77d8e1a 100755
--- a/integration/scripts/files/elements/ubuntu-trusty-guest/install.d/20-etc
+++ b/integration/scripts/files/elements/ubuntu-trusty-guest/install.d/20-etc
@@ -4,5 +4,5 @@
 # PURPOSE: take "staged" trove-guest.conf file and put it in the init directory on guest image
 
 dd if=/tmp/in_target.d/trove-guest.conf of=/etc/init/trove-guest.conf
-
+chmod a+r /etc/init/trove-guest.conf
 
diff --git a/integration/scripts/files/elements/ubuntu-trusty-guest/install.d/70-trove-guestagent b/integration/scripts/files/elements/ubuntu-trusty-guest/install.d/70-trove-guestagent
new file mode 100755
index 0000000..4e909de
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-trusty-guest/install.d/70-trove-guestagent
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# CONTEXT: GUEST during CONSTRUCTION as ROOT
+# PURPOSE: Place "staged" trove-guestagent code in the user account
+#          where upstart expects it in order to avoid downloading it
+#          when the service is started.  The code is installed under
+#          the ubuntu account which is created by script 50-user, so
+#          this script is given a higher sequence number, 70-trove.
+
+set -e
+set -o xtrace
+set -x
+
+TMP_HOOKS_PATH=/tmp/in_target.d
+
+apt-get -y install git build-essential libxml2-dev libxslt1-dev libmysqlclient-dev \
+    python-dev python-setuptools python-pip python-pexpect python-pymysql
+
+# Copy in the trove stack
+cp -r $TMP_HOOKS_PATH/trove /home/$GUEST_USERNAME
+
+# Install trove
+pushd /home/${GUEST_USERNAME}/trove
+pip install -r requirements.txt
+python setup.py install
+popd
+
+# Create the log directories.  Both in case user does not update conf file
+mkdir -p /var/log/trove $GUEST_LOGDIR
+chown -R $GUEST_USERNAME:root /var/log/trove $GUEST_LOGDIR
+chmod -R g+w /var/log/trove $GUEST_LOGDIR
+
+# Copy the trove-guestagent.conf from the controller
+mkdir -p /etc/trove/conf.d
+cp $TMP_HOOKS_PATH/trove-guestagent.conf /etc/trove/conf.d
+chown -R $GUEST_USERNAME:root /etc/trove
+chmod -R g+w /etc/trove/
+
+# Set a sentinel file to avoid rsync calls in systemd
+touch /home/$GUEST_USERNAME/trove-installed
+chown -R $GUEST_USERNAME:root /home/$GUEST_USERNAME/trove /home/$GUEST_USERNAME/trove-installed
+chmod -R g+w /home/$GUEST_USERNAME/trove /home/$GUEST_USERNAME/trove-installed
+
diff --git a/integration/scripts/files/elements/ubuntu-xenial-guest/extra-data.d/70-trove-guestagent b/integration/scripts/files/elements/ubuntu-xenial-guest/extra-data.d/70-trove-guestagent
new file mode 100755
index 0000000..85233a9
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-xenial-guest/extra-data.d/70-trove-guestagent
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
+set -o xtrace
+
+# CONTEXT: HOST prior to IMAGE BUILD as SCRIPT USER
+# PURPOSE: stages the trove github repo and trove guestagent conf so that
+#          these items can be placed in the image so that systemd does not
+#          rsync them from a controller
+
+source $_LIB/die
+
+[ -n "$TMP_HOOKS_PATH" ] || die "Temp hook path not set"
+
+[ -n "${HOST_USERNAME}" ] || die "HOST_USERNAME needs to be set to the user for the current user on the host"
+[ -n "${PATH_TROVE}" ] || die "PATH_TROVE needs to be set to the path to the trove directory on the trovestack host"
+
+cp /home/${HOST_USERNAME}/trove-guestagent.conf ${TMP_HOOKS_PATH}
+cp -r ${PATH_TROVE} ${TMP_HOOKS_PATH}
+
diff --git a/integration/scripts/files/elements/ubuntu-xenial-guest/install.d/20-etc b/integration/scripts/files/elements/ubuntu-xenial-guest/install.d/20-etc
index 1a35015..e27ad9c 100755
--- a/integration/scripts/files/elements/ubuntu-xenial-guest/install.d/20-etc
+++ b/integration/scripts/files/elements/ubuntu-xenial-guest/install.d/20-etc
@@ -1,10 +1,14 @@
-#!/bin/sh
-
+#!/bin/bash
 # CONTEXT: GUEST during CONSTRUCTION as ROOT
-# PURPOSE: take "staged" trove-guest.conf file and put it in the init directory on guest image
-
-dd if=/tmp/in_target.d/trove-guest.service of=/etc/systemd/system/trove-guest.service
+# PURPOSE: Place the "staged" trove-guestagent.conf file in the guest image
+#          and enable debug if specified during image building
 
-systemctl enable trove-guest.service
+file=/lib/systemd/system/trove-guest.service
 
+dd if=/tmp/in_target.d/trove-guest.service of=$file
 
+if [ "$DIB_MYDEBUG" == "true" ]; then
+    sed -i "s/ExecStartPre=GUEST_DEBUG/ExecStartPre=\/home\/$GUEST_USERNAME\/trove-guest-enable-debug.sh/" $file
+else
+    sed -i "s/ExecStartPre=GUEST_DEBUG/ExecStartPre=\/bin\/true/" $file
+fi
diff --git a/integration/scripts/files/elements/ubuntu-xenial-guest/install.d/70-trove-guestagent b/integration/scripts/files/elements/ubuntu-xenial-guest/install.d/70-trove-guestagent
new file mode 100755
index 0000000..a4663f8
--- /dev/null
+++ b/integration/scripts/files/elements/ubuntu-xenial-guest/install.d/70-trove-guestagent
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# CONTEXT: GUEST during CONSTRUCTION as ROOT
+# PURPOSE: Place "staged" trove-guestagent code in the user account
+#          where systemd expects it in order to avoid downloading it
+#          when the service is started.  The code is installed under
+#          the ubuntu account which is created by script 50-user, so
+#          this script is given a higher sequence number, 70-trove.
+
+set -e
+set -o xtrace
+set -x
+
+TMP_HOOKS_PATH=/tmp/in_target.d
+
+apt-get -y install git build-essential libxml2-dev libxslt1-dev libmysqlclient-dev \
+    python-dev python-setuptools python-pip python-pexpect python-pymysql
+
+# Copy in the trove stack
+cp -r $TMP_HOOKS_PATH/trove /home/$GUEST_USERNAME
+
+# Install trove
+pushd /home/${GUEST_USERNAME}/trove
+pip install -r requirements.txt
+python setup.py install
+popd
+
+# Create the log directories.  Both in case user does not update conf file
+mkdir -p /var/log/trove $GUEST_LOGDIR
+chown -R $GUEST_USERNAME:root /var/log/trove $GUEST_LOGDIR
+chmod -R g+w /var/log/trove $GUEST_LOGDIR
+
+# Copy the trove-guestagent.conf from the controller
+mkdir -p /etc/trove/conf.d
+cp $TMP_HOOKS_PATH/trove-guestagent.conf /etc/trove/conf.d
+chown -R $GUEST_USERNAME:root /etc/trove
+chmod -R g+w /etc/trove/
+
+# Set a sentinel file to avoid rsync calls in systemd
+touch /home/$GUEST_USERNAME/trove-installed
+chown -R $GUEST_USERNAME:root /home/$GUEST_USERNAME/trove /home/$GUEST_USERNAME/trove-installed
+chmod -R g+w /home/$GUEST_USERNAME/trove /home/$GUEST_USERNAME/trove-installed
+
+systemctl enable trove-guest.service
+
diff --git a/integration/scripts/files/trove-guest.systemd.conf b/integration/scripts/files/trove-guest.systemd.conf
index 47bc2cf..2a3a6ad 100644
--- a/integration/scripts/files/trove-guest.systemd.conf
+++ b/integration/scripts/files/trove-guest.systemd.conf
@@ -2,28 +2,30 @@
 Description=Trove Guest
 After=syslog.target
 After=network.target
+After=ssh.target
 
 [Service]
 Type=simple
 User=GUEST_USERNAME
 Group=GUEST_USERNAME
 
-ExecStartPre=/bin/bash -c "sudo mkdir -p GUEST_LOGDIR ; sudo chown GUEST_USERNAME:root GUEST_LOGDIR"
+# Create default log directory if it doesn't exist
+ExecStartPre=/usr/bin/sudo /bin/bash -c "mkdir -p /var/log/trove && chown GUEST_USERNAME:root /var/log/trove && chmod ug+rw,o+r /var/log/trove"
 
-# If ~/trove-installed does not exist, copy the trove source from
-# the user's development environment, then touch the sentinel file
-ExecStartPre=/bin/bash -c "test -d /home/GUEST_USERNAME/trove-installed || sudo -u GUEST_USERNAME rsync -e 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no' -avz --exclude='.*' HOST_SCP_USERNAME@CONTROLLER_IP:PATH_TROVE/ /home/GUEST_USERNAME/trove && touch /home/GUEST_USERNAME/trove-installed"
+# guest_info and trove-guestagent.conf are injected.  Make sure all files are readable
+ExecStartPre=/usr/bin/sudo /bin/bash -c "test ! -d /etc/trove || chown -R GUEST_USERNAME:root /etc/trove && chmod -R ug+rw,o+r /etc/trove"
 
-# If /etc/trove does not exist, create it and then copy the trove-guestagent.conf
-# from /etc/trove on the user's development environment,
-ExecStartPre=/bin/bash -c "test -d /etc/trove/conf.d || sudo mkdir -p /etc/trove/conf.d && sudo -u GUEST_USERNAME rsync -e 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no' -avz --exclude='.*' HOST_SCP_USERNAME@CONTROLLER_IP:/etc/trove/trove-guestagent.conf ~GUEST_USERNAME/ && sudo mv ~GUEST_USERNAME/trove-guestagent.conf /etc/trove/conf.d/trove-guestagent.conf"
-
-ExecStartPre=/bin/bash -c "sudo chown -R GUEST_USERNAME:root /etc/trove"
+# Optionally set debug logging in trove-guestagent
+ExecStartPre=GUEST_DEBUG
 
 ExecStart=/home/GUEST_USERNAME/trove/contrib/trove-guestagent --config-dir=/etc/trove/conf.d
 
 # Give a reasonable amount of time for the server to start up/shut down
+Restart=on-failure
 TimeoutSec=300
+TimeoutStopSec=15
+
+LimitNOFILE=65535
 
 # PgSql doesn't play nice with PrivateTmp
 PrivateTmp=false
diff --git a/integration/scripts/files/elements/ubuntu-guest/install.d/50-user b/integration/scripts/files/elements/ubuntu-guest/install.d/50-user
deleted file mode 100755
index 99f6896..0000000
--- a/integration/scripts/files/elements/ubuntu-guest/install.d/50-user
+++ /dev/null
@@ -1,18 +0,0 @@
-#!/bin/bash
-
-# CONTEXT: GUEST during CONSTRUCTION as ROOT
-# PURPOSE: Add the guest image user that will own the trove agent source...if the user does not already exist
-
-set -e
-set -o xtrace
-
-if ! id -u ${GUEST_USERNAME} >/dev/null 2>&1; then
-    echo "Adding ${GUEST_USERNAME} user"
-    useradd -G sudo -m ${GUEST_USERNAME} -s /bin/bash
-    chown ${GUEST_USERNAME}:${GUEST_USERNAME} /home/${GUEST_USERNAME}
-    passwd ${GUEST_USERNAME} <<_EOF_
-${GUEST_USERNAME}
-${GUEST_USERNAME}
-_EOF_
-fi
-
