diff --git a/integration/scripts/files/elements/ubuntu-postgresql/install.d/30-postgresql b/integration/scripts/files/elements/ubuntu-postgresql/install.d/30-postgresql
index d5b5f50..630886b 100755
--- a/integration/scripts/files/elements/ubuntu-postgresql/install.d/30-postgresql
+++ b/integration/scripts/files/elements/ubuntu-postgresql/install.d/30-postgresql
@@ -31,48 +31,7 @@ exit \$?
 
 _EOF_
 
-apt-get --allow-unauthenticated -y install postgresql-9.4 postgresql-contrib-9.4 postgresql-server-dev-9.4
-
-###########################################
-# Hack alert:
-# For Postgresql 9.4, pg_rewind is not in the main source tree and
-# no packages exist in the repos, so it must be compiled manually
-# and installed on the image until we can move to 9.5
-# See README at
-# https://github.com/vmware/pg_rewind/tree/REL9_4_STABLE
-
-tmpdir=/tmp/build
-mkdir -p $tmpdir
-cd $tmpdir
-git clone https://github.com/postgres/postgres.git --branch REL9_4_STABLE
-cd postgres/contrib
-git clone https://github.com/vmware/pg_rewind.git --branch REL9_4_STABLE
-
-dev_pkgs="libreadline-dev libkrb5-dev libssl-dev libpam-dev libxml2-dev libxslt-dev libedit-dev libselinux1-dev bison flex"
-
-apt-get --allow-unauthenticated install $dev_pkgs -y
-
-# Unfortunately, on ubuntu, was not able to get pg_rewind to build
-# outside of the pgsql source tree. Configure and compile postgres
-# but only call make install against the contrib/pg_rewind directory
-# so that support library is accessible to the server
-cd $tmpdir/postgres
-./configure
-make
-cd contrib/pg_rewind
-make install
-
-# Make the pg_rewind binary and the library used by the
-# pg_rewind stored procedures accessible
-ln -s /usr/local/pgsql/bin/pg_rewind /usr/bin/pg_rewind
-ln -s /usr/local/pgsql/lib/pg_rewind_support.so /usr/lib/postgresql/9.4/lib/pg_rewind_support.so
-
-cd
-rm -rf $tmpdir
-apt-get remove -y $dev_pkgs
-
-# End hack
-################################
+apt-get --allow-unauthenticated -y install postgresql postgresql-contrib postgresql-client
 
 # Install the native Python client.
 apt-get --allow-unauthenticated -y install libpq-dev
diff --git a/integration/scripts/files/elements/ubuntu-postgresql/pre-install.d/10-postgresql-repo b/integration/scripts/files/elements/ubuntu-postgresql/pre-install.d/10-postgresql-repo
index 132bc4f..2bc4f15 100755
--- a/integration/scripts/files/elements/ubuntu-postgresql/pre-install.d/10-postgresql-repo
+++ b/integration/scripts/files/elements/ubuntu-postgresql/pre-install.d/10-postgresql-repo
@@ -1,10 +1,14 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 set -o xtrace
 
 [ -n "${RELEASE}" ] || die "RELEASE must be set to a valid Ubuntu release (e.g. trusty)"
 
+if [ "$DIB_MYCOMMUNITY_EDITION" == "false" ]; then
+    exit 0
+fi
+
 cat <<EOL > /etc/apt/sources.list.d/postgresql.list
 deb http://apt.postgresql.org/pub/repos/apt/ ${RELEASE}-pgdg main
 EOL
diff --git a/integration/scripts/files/elements/ubuntu-xenial-postgresql/install.d/31-fix-init-script b/integration/scripts/files/elements/ubuntu-xenial-postgresql/install.d/31-fix-init-script
index c454d83..bde587b 100755
--- a/integration/scripts/files/elements/ubuntu-xenial-postgresql/install.d/31-fix-init-script
+++ b/integration/scripts/files/elements/ubuntu-xenial-postgresql/install.d/31-fix-init-script
@@ -1,7 +1,8 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 set -o xtrace
 
-sed -i "s/PIDFile=\/var\/run\/postgresql\/%i.pid/PIDFile=\/var\/run\/postgresql\/postgresql.pid/" /lib/systemd/system/postgresql@.service
-
+if [ "$DIB_MYCOMMUNITY_EDITION" == "true" ]; then
+    sed -i "s/PIDFile=\/var\/run\/postgresql\/%i.pid/PIDFile=\/var\/run\/postgresql\/postgresql.pid/" /lib/systemd/system/postgresql@.service
+fi
