diff --git a/trove/guestagent/datastore/experimental/mariadb/service.py b/trove/guestagent/datastore/experimental/mariadb/service.py
index a851e02..6e70998 100644
--- a/trove/guestagent/datastore/experimental/mariadb/service.py
+++ b/trove/guestagent/datastore/experimental/mariadb/service.py
@@ -36,6 +36,7 @@ class MariaDBApp(galera_service.GaleraApp):
     @property
     def service_candidates(self):
         service_candidates = super(MariaDBApp, self).service_candidates
+        LOG.debug("MariaDBApp services_candidates=%s" % service_candidates)
         return {
             operating_system.DEBIAN: ["mariadb"] + service_candidates,
             operating_system.REDHAT: ["mariadb"],
@@ -54,6 +55,7 @@ class MariaDBApp(galera_service.GaleraApp):
             result['cmd_bootstrap_galera_cluster'] = (
                 "sudo systemctl start %s@bootstrap.service"
                 % result['service'])
+        LOG.debug("mysql_service: result=%s" % str(result))
         return result
 
     @property
@@ -99,7 +101,6 @@ class MariaDBRootAccess(mysql_service.BaseMySqlRootAccess):
             mysql_service.BaseLocalSqlClient,
             MariaDBApp(mysql_service.BaseMySqlAppStatus.get()))
 
-
 class MariaDBAdmin(mysql_service.BaseMySqlAdmin):
     def __init__(self):
         super(MariaDBAdmin, self).__init__(
diff --git a/trove/guestagent/datastore/mysql_common/manager.py b/trove/guestagent/datastore/mysql_common/manager.py
index 54b3c4a..31d48ac 100644
--- a/trove/guestagent/datastore/mysql_common/manager.py
+++ b/trove/guestagent/datastore/mysql_common/manager.py
@@ -85,6 +85,8 @@ class MySqlManager(manager.Manager):
         slow_query_log_file = self.build_log_file_name(
             self.GUEST_LOG_DEFS_SLOW_QUERY_LABEL, owner,
             datastore_dir=datastore_dir)
+        LOG.debug("datastore_log_defs: datastore_dir=%s, general_log=%s, error_log=%s, slow_query_log=%s" %
+                  (datastore_dir, error_log_file, general_log_file, slow_query_log_file))
         return {
             self.GUEST_LOG_DEFS_GENERAL_LABEL: {
                 self.GUEST_LOG_TYPE_LABEL: guest_log.LogType.USER,
@@ -200,6 +202,10 @@ class MySqlManager(manager.Manager):
                    config_contents, root_password, overrides,
                    cluster_config, snapshot):
         """This is called from prepare in the base class."""
+
+        LOG.debug("do_prepare: device_path=%s, packages=%s, backup_info=%s, cluster_config=%s" %
+                  (device_path, packages, str(backup_info), str(cluster_config)))
+
         app = self.mysql_app(self.mysql_app_status.get())
         app.install_if_needed(packages)
         if device_path:
