diff --git a/integration/scripts/files/elements/ubuntu-guest/install.d/15-trove-dep b/integration/scripts/files/elements/ubuntu-guest/install.d/15-trove-dep
index c79ad87..050ffa7 100755
--- a/integration/scripts/files/elements/ubuntu-guest/install.d/15-trove-dep
+++ b/integration/scripts/files/elements/ubuntu-guest/install.d/15-trove-dep
@@ -24,7 +24,7 @@ if [ -f ${TMP_HOOKS_DIR}/upper-constraints.txt ]; then
     UPPER_CONSTRAINTS=" -c ${TMP_HOOKS_DIR}/upper-constraints.txt"
 fi
 
-pip2 install -q --upgrade -r ${TMP_HOOKS_DIR}/requirements.txt ${UPPER_CONSTRAINTS}
+pip2 install -q --upgrade -r ${TMP_HOOKS_DIR}/requirements.txt
 
 echo "diagnostic pip freeze output follows"
 pip2 freeze
