diff -Naur a/opt/ansible_v1.9.4-1/lib/ansible/runner/__init__.py b/opt/ansible_v1.9.4-1/lib/ansible/runner/__init__.py
--- a/opt/ansible_v1.9.4-1/lib/ansible/runner/__init__.py	2016-06-08 14:33:02.462341477 -0500
+++ b/opt/ansible_v1.9.4-1/lib/ansible/runner/__init__.py	2016-06-08 14:24:06.815390332 -0500
@@ -1383,11 +1383,13 @@
             fileno = None
 
         workers = []
+        fdes = []
         for i in range(self.forks):
             new_stdin = None
             if fileno is not None:
                 try:
                     new_stdin = os.fdopen(os.dup(fileno))
+                    fdes.append(new_stdin)
                 except OSError, e:
                     # couldn't dupe stdin, most likely because it's
                     # not a valid file descriptor, so we just rely on
@@ -1405,6 +1407,9 @@
             for worker in workers:
                 worker.terminate()
                 worker.join()
+        finally:
+            for f in fdes:
+                f.close()
 
         results = []
         try:
