diff -Naur a/etc/ansible/roles/os_trove/templates/trove-guestagent.conf.j2 b/etc/ansible/roles/os_trove/templates/trove-guestagent.conf.j2
--- a/etc/ansible/roles/os_trove/templates/trove-guestagent.conf.j2	2017-02-14 16:05:00.019587342 -0600
+++ b/etc/ansible/roles/os_trove/templates/trove-guestagent.conf.j2	2017-02-14 16:04:24.967585757 -0600
@@ -7,13 +7,12 @@
 nova_proxy_admin_pass = {{ trove_service_password }}
 nova_proxy_admin_tenant_name = {{ trove_service_project_name }}
 
-{% if trove_swift_enabled is defined %}
-swift_url = {{ trove_swift_url }}
+# Information to get Swift URL from Keystone
+swift_endpoint_type = internalURL
 # Region name of this node. Default value is None.
 os_region_name = {{ trove_service_region }}
 # Service type to use when searching catalog.
 swift_service_type = object-store
-{% endif %}
 
 # ========== Datastore Manager Configurations ==========
 # Datastore manager implementations.
@@ -55,19 +54,19 @@
 [oslo_messaging_rabbit]
 # The RabbitMQ broker address where a single node is used. (string value)
 # Deprecated group/name - [DEFAULT]/rabbit_host
-rabbit_host = {{ trove_rabbitmq_servers }}
+#rabbit_host = {{ trove_rabbitmq_servers }}
 
 # The RabbitMQ broker port where a single node is used. (integer value)
 # Deprecated group/name - [DEFAULT]/rabbit_port
-rabbit_port= {{ trove_rabbitmq_port }}
+#rabbit_port= {{ trove_rabbitmq_port }}
 
 # RabbitMQ HA cluster host:port pairs. (list value)
 # Deprecated group/name - [DEFAULT]/rabbit_hosts
-#rabbit_hosts=$rabbit_host:$rabbit_port
+rabbit_hosts={% for host in trove_rabbitmq_servers.split(',') %}{{ host }}:{{ trove_rabbitmq_port }}{% if not loop.last %},{% endif %}{% endfor %}
 
 # Connect over SSL for RabbitMQ. (boolean value)
 # Deprecated group/name - [DEFAULT]/rabbit_use_ssl
-#rabbit_use_ssl=false
+rabbit_use_ssl = {{ trove_rabbitmq_use_ssl }}
 
 # The RabbitMQ userid. (string value)
 # Deprecated group/name - [DEFAULT]/rabbit_userid
