diff -Naur a/etc/ansible/roles/os_neutron/templates/metering_agent.ini.j2 b/etc/ansible/roles/os_neutron/templates/metering_agent.ini.j2
--- a/etc/ansible/roles/os_neutron/templates/metering_agent.ini.j2	2017-02-08 17:17:27.021853224 -0600
+++ b/etc/ansible/roles/os_neutron/templates/metering_agent.ini.j2	2017-02-08 17:15:43.753229595 -0600
@@ -6,7 +6,15 @@
 
 # Drivers
 driver = {{ neutron_driver_metering }}
-interface_driver = {{ neutron_plugins[neutron_plugin_type].driver_interface }}
+
+# Other agents will use neutron_plugins[neutron_plugin_type].driver_interface
+# for the interface_driver, but that uses a stevedore alias. Metering agent
+# hasn't been updated to use stevedore alises so that fails. To work around
+# the problem until metering agent is updated, we should use the full
+# module.class path to the interface driver.
+# TODO(hughsaunders): switch back to stevedore when
+# https://review.openstack.org/#/c/419881/ merges and is backported.
+interface_driver = {{ neutron_services['neutron-metering-agent'].interface_driver }}
 
 # Intervals
 measure_interval = 30
