diff -Naur a/etc/ansible/roles/repo_build/templates/op-venv-script.sh.j2 b/etc/ansible/roles/repo_build/templates/op-venv-script.sh.j2
--- a/etc/ansible/roles/repo_build/templates/op-venv-script.sh.j2	2016-06-14 15:14:45.000000000 -0500
+++ b/etc/ansible/roles/repo_build/templates/op-venv-script.sh.j2	2016-06-14 16:46:07.000000000 -0500
@@ -61,8 +61,15 @@
 #  If the venv archive already exists the creation process will be skipped
 pushd "{{ repo_build_venv_dir }}/{{ repo_build_release_tag }}/{{ ansible_distribution | lower }}"
 {% for key, value in os_group.items() %}
+
 ROLE_VENV_PATH="{{ repo_build_venv_build_dir }}/venvs/{{ key | replace('os_', '') }}"
-ROLE_VENV_FILE="{{ key | replace('os_', '') }}-{{ repo_build_release_tag }}"
+
+if [ "{{ ansible_architecture }}" = "ppc64le" ];then
+    ROLE_VENV_FILE="{{ key | replace('os_', '') }}-{{ repo_build_release_tag }}-{{ ansible_architecture }}"
+else
+    ROLE_VENV_FILE="{{ key | replace('os_', '') }}-{{ repo_build_release_tag }}"
+fi
+
 if [ ! -f "${ROLE_VENV_FILE}.tgz" ];then
     venv_create "${ROLE_VENV_PATH}" "${ROLE_VENV_FILE}" "{{ value | join(' ') }}" &
     pid[{{ loop.index }}]=$!
