diff -Naur a/etc/ansible/roles/os_heat/templates/heat.conf.j2 b/etc/ansible/roles/os_heat/templates/heat.conf.j2
--- a/etc/ansible/roles/os_heat/templates/heat.conf.j2	2017-01-23 00:54:56.965935822 -0500
+++ b/etc/ansible/roles/os_heat/templates/heat.conf.j2	2017-01-23 00:53:27.425934221 -0500
@@ -66,6 +66,11 @@
 [oslo_messaging_rabbit]
 rabbit_use_ssl = {{ rabbitmq_use_ssl }}
 
+{% if heat_service_publicuri_proto != heat_service_internaluri_proto %}
+[oslo_middleware]
+enable_proxy_headers_parsing = True
+{% endif %}
+
 # Ceilometer options
 {% if heat_ceilometer_enabled %}
 [oslo_messaging_notifications]
