diff -Naur a/etc/ansible/roles/os_neutron/templates/dhcp_agent.ini.j2 b/etc/ansible/roles/os_neutron/templates/dhcp_agent.ini.j2
--- a/etc/ansible/roles/os_neutron/templates/dhcp_agent.ini.j2	2017-02-09 01:13:28.856219895 -0500
+++ b/etc/ansible/roles/os_neutron/templates/dhcp_agent.ini.j2	2017-02-09 00:49:46.272194457 -0500
@@ -1,6 +1,6 @@
 # {{ ansible_managed }}
 
-{% set _api_threads = ansible_processor_vcpus|default(2) // 2 %}
+{% set _api_threads = [ansible_processor_vcpus|default(2) // 2, neutron_api_threads_max] | min %}
 {% set api_threads = _api_threads if _api_threads > 0 else 1 %}
 
 # General
