diff -Naur a/etc/ansible/roles/os_trove/templates/trove.conf.j2 b/etc/ansible/roles/os_trove/templates/trove.conf.j2
--- a/etc/ansible/roles/os_trove/templates/trove.conf.j2	2017-02-09 01:13:25.200219829 -0500
+++ b/etc/ansible/roles/os_trove/templates/trove.conf.j2	2017-02-09 00:56:24.772201583 -0500
@@ -1,5 +1,5 @@
 # {{ ansible_managed }}
-{% set _api_threads = ansible_processor_vcpus|default(2) // 2 %}
+{% set _api_threads = [ansible_processor_vcpus|default(2) // 2, trove_api_threads_max] | min %}
 {% set api_threads = _api_threads if _api_threads > 0 else 1 %}
 [DEFAULT]
 debug = {{ debug }}
