diff -Naur a/etc/ansible/roles/os_trove/templates/trove.conf.j2 b/etc/ansible/roles/os_trove/templates/trove.conf.j2
--- a/etc/ansible/roles/os_trove/templates/trove.conf.j2	2017-06-29 05:40:27.953615133 -0400
+++ b/etc/ansible/roles/os_trove/templates/trove.conf.j2	2017-06-29 03:09:06.929452752 -0400
@@ -29,6 +29,9 @@
 max_backups_per_tenant = 5
 volume_time_out=30
 
+# Swift URL
+swift_url = {{ trove_swift_url }}
+
 # Config options for rate limits
 http_get_rate = 200
 http_post_rate = 200
@@ -57,8 +60,8 @@
 nova_proxy_admin_tenant_name = {{ trove_service_project_name }}
 
 # Neutron
-network_driver = trove.network.nova.NovaNetwork
-default_neutron_networks =
+network_driver = trove.network.neutron.NeutronDriver
+default_neutron_networks = {{ trove_infra_net_id }}
 
 # Taskmanager queue name
 taskmanager_queue = taskmanager
@@ -128,7 +131,7 @@
 [redis]
 tcp_ports = 6379
 #redis uses local storage
-volume_support = False
+volume_support = True
 # default device_path = None
 
 [cassandra]
