diff -Naur a/etc/ansible/roles/os_heat/templates/heat.conf.j2 b/etc/ansible/roles/os_heat/templates/heat.conf.j2
--- a/etc/ansible/roles/os_heat/templates/heat.conf.j2	2017-03-02 10:44:52.058156762 -0500
+++ b/etc/ansible/roles/os_heat/templates/heat.conf.j2	2017-03-02 05:48:29.101838778 -0500
@@ -1,6 +1,6 @@
 # {{ ansible_managed }}
 
-{% set _api_threads = ansible_processor_vcpus|default(2) // 2 %}
+{% set _api_threads = [ansible_processor_vcpus|default(2) // 2, heat_api_threads_max] | min %}
 {% set api_threads = _api_threads if _api_threads > 0 else 1 %}
 
 [DEFAULT]
