diff -Naur a/etc/ansible/plugins/library/provider_networks b/etc/ansible/plugins/library/provider_networks
--- a/etc/ansible/plugins/library/provider_networks	2016-11-21 08:54:54.733921686 -0600
+++ b/etc/ansible/plugins/library/provider_networks	2016-11-21 08:46:17.504688259 -0600
@@ -192,9 +192,9 @@
             elif net['network']['type'] == "flat":
                 if "flat" not in self.network_types:
                     self.network_types.append('flat')
-                    self.network_flat_networks.append(
-                        net['network']['net_name']
-                    )
+                self.network_flat_networks.append(
+                    net['network']['net_name']
+                )
 
             # Create the network mappings
             if net['network']['type'] not in ['raw', 'vxlan']:
