diff -Naur a/opt/ansible_v1.9.4-1/lib/ansible/module_utils/basic.py b/opt/ansible_v1.9.4-1/lib/ansible/module_utils/basic.py
--- a/opt/ansible_v1.9.4-1/lib/ansible/module_utils/basic.py	2016-06-17 16:15:29.464030214 -0500
+++ b/opt/ansible_v1.9.4-1/lib/ansible/module_utils/basic.py	2016-06-17 09:15:09.079223159 -0500
@@ -1569,14 +1569,15 @@
                     # left, there is no need to call select() again.
                     break
 
-            cmd.stdout.close()
-            cmd.stderr.close()
-
             rc = cmd.returncode
+
         except (OSError, IOError), e:
             self.fail_json(rc=e.errno, msg=str(e), cmd=clean_args)
         except:
             self.fail_json(rc=257, msg=traceback.format_exc(), cmd=clean_args)
+        finally:
+            cmd.stdout.close()
+            cmd.stderr.close()
 
         if rc != 0 and check_rc:
             msg = heuristic_log_sanitize(stderr.rstrip())
