diff -Naur a/opt/ansible_v1.9.4-1/lib/ansible/module_utils/facts.py b/opt/ansible_v1.9.4-1/lib/ansible/module_utils/facts.py
--- a/opt/ansible_v1.9.4-1/lib/ansible/module_utils/facts.py	2016-09-29 20:30:24.772322000 +0000
+++ b/opt/ansible_v1.9.4-1/lib/ansible/module_utils/facts.py	2016-09-29 18:38:56.402322002 +0000
@@ -762,7 +762,7 @@
 
             # model name is for Intel arch, Processor (mind the uppercase P)
             # works for some ARM devices, like the Sheevaplug.
-            if key == 'model name' or key == 'Processor' or key == 'vendor_id':
+            if key in ['model name', 'Processor', 'vendor_id', 'cpu', 'Vendor']:
                 if 'processor' not in self.facts:
                     self.facts['processor'] = []
                 self.facts['processor'].append(data[1].strip())
@@ -786,8 +786,10 @@
             elif key == '# processors':
                 self.facts['processor_cores'] = int(data[1].strip())
 
-        if vendor_id_occurrence == model_name_occurrence:
-            i = vendor_id_occurrence
+        # Skip for platforms without vendor_id/model_name in cpuinfo (e.g ppc64le)
+        if vendor_id_occurrence > 0:
+            if vendor_id_occurrence == model_name_occurrence:
+                i = vendor_id_occurrence
 
         if self.facts['architecture'] != 's390x':
             if xen_paravirt:
