diff -Naur a/opt/openstack-ansible/tox.ini b/opt/openstack-ansible/tox.ini
--- a/opt/openstack-ansible/tox.ini	2017-01-25 11:49:44.717728247 -0500
+++ b/opt/openstack-ansible/tox.ini	2017-01-25 11:34:59.581712419 -0500
@@ -8,8 +8,9 @@
 usedevelop = True
 basepython = python2.7
 install_command =
-    pip install -c{env:UPPER_CONSTRAINTS_FILE:https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=stable/newton} {opts} {packages}
+    pip install -c{toxinidir}/global-requirement-pins.txt -c{env:UPPER_CONSTRAINTS_FILE:https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=stable/newton} {opts} {packages}
 deps =
+    -r{toxinidir}/global-requirement-pins.txt
     -r{toxinidir}/test-requirements.txt
     sphinxmark>=0.1.14
 passenv =
