diff -Naur a/opt/openstack-ansible/scripts/bootstrap-ansible.sh b/opt/openstack-ansible/scripts/bootstrap-ansible.sh
--- a/opt/openstack-ansible/scripts/bootstrap-ansible.sh	2017-01-25 11:49:44.713728247 -0500
+++ b/opt/openstack-ansible/scripts/bootstrap-ansible.sh	2017-01-25 11:35:25.593712884 -0500
@@ -22,7 +22,7 @@
 ## Vars ----------------------------------------------------------------------
 export HTTP_PROXY=${HTTP_PROXY:-""}
 export HTTPS_PROXY=${HTTPS_PROXY:-""}
-export ANSIBLE_PACKAGE=${ANSIBLE_PACKAGE:-"ansible==2.1.4.0"}
+export ANSIBLE_PACKAGE=${ANSIBLE_PACKAGE:-"git+https://github.com/ansible/ansible@44026f8d7befa4d25c8ee8b3bd3efa3bdb240d62"} # stable-2.1 post 2.1.4.0 with yaml fix
 export ANSIBLE_ROLE_FILE=${ANSIBLE_ROLE_FILE:-"ansible-role-requirements.yml"}
 export SSH_DIR=${SSH_DIR:-"/root/.ssh"}
 export DEBIAN_FRONTEND=${DEBIAN_FRONTEND:-"noninteractive"}
@@ -110,7 +110,7 @@
 ${PIP_COMMAND} install ${PIP_OPTS} ${PIP_INSTALL_OPTIONS} || ${PIP_COMMAND} install ${PIP_OPTS} --isolated ${PIP_INSTALL_OPTIONS}
 
 # Set the constraints now that we know we're using the right version of pip
-PIP_OPTS+=" --constraint ${UPPER_CONSTRAINTS_FILE}"
+PIP_OPTS+=" --constraint global-requirement-pins.txt --constraint ${UPPER_CONSTRAINTS_FILE}"
 
 # Install the required packages for ansible
 $PIP_COMMAND install $PIP_OPTS -r requirements.txt ${ANSIBLE_PACKAGE} || $PIP_COMMAND install --isolated $PIP_OPTS -r requirements.txt ${ANSIBLE_PACKAGE}
