diff -Naur a/opt/ansible_v1.9.4-1/lib/ansible/cache/jsonfile.py b/opt/ansible_v1.9.4-1/lib/ansible/cache/jsonfile.py
--- a/opt/ansible_v1.9.4-1/lib/ansible/cache/jsonfile.py	2016-06-25 06:27:06.700581645 -0400
+++ b/opt/ansible_v1.9.4-1/lib/ansible/cache/jsonfile.py	2016-06-25 06:19:00.952563891 -0400
@@ -73,6 +73,7 @@
         self._cache[key] = value
 
         cachefile = "%s/%s" % (self._cache_dir, key)
+        f = None
         try:
             f = codecs.open(cachefile, 'w', encoding='utf-8')
         except (OSError,IOError), e:
@@ -80,7 +81,8 @@
         else:
             f.write(utils.jsonify(value))
         finally:
-            f.close()
+            if f:
+                f.close()
 
     def has_expired(self, key):
 
