diff -Naur a/opt/ansible_v1.9.4-1/lib/ansible/modules/core/packaging/os/apt.py b/opt/ansible_v1.9.4-1/lib/ansible/modules/core/packaging/os/apt.py
--- a/opt/ansible_v1.9.4-1/lib/ansible/modules/core/packaging/os/apt.py	2016-11-18 22:09:30.492269528 -0600
+++ b/opt/ansible_v1.9.4-1/lib/ansible/modules/core/packaging/os/apt.py	2016-11-18 22:11:43.633004127 -0600
@@ -588,7 +588,16 @@
                         cache_valid = True
 
             if cache_valid is not True:
-                cache.update()
+                # Retry to update the cache up to 3 times
+                for retry in range(3):
+                    try:
+                        cache.update()
+                        break
+                    except apt.cache.FetchFailedException:
+                        pass
+                else:
+                    module.fail_json(msg='Failed to update apt cache. Retries exhausted.')
+
                 cache.open(progress=None)
             if not p['package'] and not p['upgrade'] and not p['deb']:
                 module.exit_json(changed=False)
