diff -Naur a/opt/ansible_v1.9.4-1/lib/ansible/utils/hashing.py b/opt/ansible_v1.9.4-1/lib/ansible/utils/hashing.py
--- a/opt/ansible_v1.9.4-1/lib/ansible/utils/hashing.py	2016-06-20 10:24:57.547351911 -0500
+++ b/opt/ansible_v1.9.4-1/lib/ansible/utils/hashing.py	2016-06-20 10:17:46.847357682 -0500
@@ -58,14 +58,14 @@
     digest = hash_func()
     blocksize = 64 * 1024
     try:
-        infile = open(filename, 'rb')
-        block = infile.read(blocksize)
-        while block:
-            digest.update(block)
+        with open(filename, 'rb') as infile:
             block = infile.read(blocksize)
-        infile.close()
+            while block:
+                digest.update(block)
+                block = infile.read(blocksize)
     except IOError, e:
         raise AnsibleError("error while accessing the file %s, error was: %s" % (filename, e))
+
     return digest.hexdigest()
 
 # The checksum algorithm must match with the algorithm in ShellModule.checksum() method
