diff --git a/trove/guestagent/datastore/mysql_common/service.py b/trove/guestagent/datastore/mysql_common/service.py
index 96c9330..0645a3f 100644
--- a/trove/guestagent/datastore/mysql_common/service.py
+++ b/trove/guestagent/datastore/mysql_common/service.py
@@ -102,7 +102,7 @@ def clear_expired_password():
     if m:
         try:
             out, err = utils.execute("mysqladmin", "-p%s" % m.group(1),
-                                     "password", "", run_as_root=True,
+                                     "password", "", "reload", run_as_root=True,
                                      root_helper="sudo")
         except exception.ProcessExecutionError:
             LOG.exception(_("Cannot change mysql password."))
