diff --git a/trove/guestagent/datastore/experimental/postgresql/pgutil.py b/trove/guestagent/datastore/experimental/postgresql/pgutil.py
index 43eb637..7ac1f18 100644
--- a/trove/guestagent/datastore/experimental/postgresql/pgutil.py
+++ b/trove/guestagent/datastore/experimental/postgresql/pgutil.py
@@ -168,7 +168,7 @@ class UserQuery(object):
         create_clause = "CREATE USER \"{name}\"".format(name=name)
         with_clause = cls._build_with_clause(
             password, encrypt_password, *options)
-        return ''.join([create_clause, with_clause])
+        return ' '.join([create_clause, with_clause])
 
     @classmethod
     def _build_with_clause(cls, password, encrypt_password=None, *options):
@@ -203,7 +203,7 @@ class UserQuery(object):
         alter_clause = "ALTER USER \"{name}\"".format(name=name)
         with_clause = cls._build_with_clause(
             password, encrypt_password, *options)
-        return ''.join([alter_clause, with_clause])
+        return ' '.join([alter_clause, with_clause])
 
     @classmethod
     def update_name(cls, old, new):
diff --git a/trove/guestagent/datastore/experimental/postgresql/service/config.py b/trove/guestagent/datastore/experimental/postgresql/service/config.py
index 644d368..878ec4e 100644
--- a/trove/guestagent/datastore/experimental/postgresql/service/config.py
+++ b/trove/guestagent/datastore/experimental/postgresql/service/config.py
@@ -221,12 +221,13 @@ class PgSqlConfig(PgSqlProcess):
             'wal_level': 'hot_standby',
             'archive_mode ': 'on',
             'max_wal_senders': 8,
-            'checkpoint_segments ': 8,
             'wal_keep_segments': 8,
             'archive_command': arch_cmd
         }
         if not self.pg_version[1] in ('9.3'):
             opts['wal_log_hints'] = 'on'
+        if self.pg_version[1] in ('9.6'):
+            opts['wal_level'] = 'replica'  # replica subsumes hot_standby which is removed
 
         self.configuration_manager.apply_system_override(
             opts, BACKUP_CFG_OVERRIDE)
diff --git a/trove/guestagent/datastore/experimental/postgresql/service/users.py b/trove/guestagent/datastore/experimental/postgresql/service/users.py
index 0c61bbe..8c8f494 100644
--- a/trove/guestagent/datastore/experimental/postgresql/service/users.py
+++ b/trove/guestagent/datastore/experimental/postgresql/service/users.py
@@ -51,6 +51,7 @@ class PgSqlUsers(PgSqlAccess):
             'CREATEROLE',
             'INHERIT',
             'REPLICATION',
+            'BYPASSRLS',
             'LOGIN']
 
     def _create_admin_user(self, context, databases=None):
diff --git a/trove/guestagent/strategies/replication/experimental/postgresql_impl.py b/trove/guestagent/strategies/replication/experimental/postgresql_impl.py
index 9942a67..e6ca988 100644
--- a/trove/guestagent/strategies/replication/experimental/postgresql_impl.py
+++ b/trove/guestagent/strategies/replication/experimental/postgresql_impl.py
@@ -280,8 +280,10 @@ class PostgresqlReplicationStreaming(
         opts = {'hot_standby': 'on',
                 'wal_level': 'hot_standby'}
         # wal_log_hints for pg_rewind is only supported in 9.4+
-        if self.pg_version[1] in ('9.4', '9.5'):
+        if self.pg_version[1] in ('9.4', '9.5', '9.6'):
             opts['wal_log_hints'] = 'on'
+        if self.pg_version[1] == '9.6':
+            opts['wal_level'] = 'replica'    # replica subsumes hot_standby which is removed
 
         service.configuration_manager.\
             apply_system_override(opts, SLAVE_STANDBY_OVERRIDE)
